fdbserver/storageserver.actor.cpp

```diff
@@ -1947,11 +1982,23 @@ ACTOR Future<Void> logFetchKeysWarning(AddingShard* shard) {
 }
 
 ACTOR Future<Void> fetchKeys( StorageServer *data, AddingShard* shard ) {
+	state UID ID = deterministicRandom()->randomUniqueID();
 	state TraceInterval interval("FetchKeys");
 	state KeyRange keys = shard->keys;
 	state Future<Void> warningLogger = logFetchKeysWarning(shard);
-	state double startt = now();
+	state const double startTime = now();
+	state int fetchedBytes = 0;
 	state int fetchBlockBytes = BUGGIFY ? SERVER_KNOBS->BUGGIFY_BLOCK_BYTES : SERVER_KNOBS->FETCH_BLOCK_BYTES;
+	state std::function<void()> reportHistogram = [this, data]() -> void {
```

Comment:
Use a knob that can disable the reporting mechanism and its overhead on the fly, in case it causes too much overhead.